src/handler.ts

```diff
@@ -25,11 +26,26 @@ export const createHandler =
     const config = await configPromise;
     const elasticClient = await elasticClientPromise;
 
+    let failures: OnDropDocument<LogDocument>[] = [];
     await elasticClient.helpers.bulk<LogDocument>({
       datasource: documents,
-      onDocument: () => ({ index: { _index: config.dataStreamName } }),
+      onDocument: (doc) => ({
+        create: { _index: config.dataStreamName, _id: doc.id },
+      }),
       onDrop: (fail) => {
-        console.warn("Failed to ingest document: ", fail);
+        failures.push(fail);
       },
     });
+
+    if (failures.length !== 0) {
+      const failedServices = new Set(
+        failures.map(({ document }) => document.service)
+      );
+      console.error(
+        `Failed to ingest documents from ${failedServices.size} services: ${[
+          ...failedServices,
+        ].join(", ")}`
```

Comment:
Why do we need the spread operator here? I don't think I understand what the difference is between this and:

```suggestion
        `Failed to ingest documents from ${failedServices.size} services: ${failedServices.join(", ")}`
```

Reply:
Because it's a Set, and needs to be spread into an array to have access to array ops